test/rssignores.cpp

```diff
@@ -39,3 +43,219 @@ TEST_CASE(
 	REQUIRE(ignores.matches_resetunread("www.cool-website.com"));
 	REQUIRE_FALSE(ignores.matches_resetunread("www.smth.com"));
 }
+
+TEST_CASE("RssIgnores::handle_action() handles `ignore-article`",
+	"[RssIgnores]")
+{
+	RssIgnores ignores;
+
+	const std::string action = "ignore-article";
+
+	SECTION("Throws ConfigHandlerException if less than 2 parameters") {
+		REQUIRE_THROWS_AS(ignores.handle_action(action, {}), ConfigHandlerException);
+		REQUIRE_THROWS_AS(ignores.handle_action(action, {"param"}),
+			ConfigHandlerException);
+	}
+
+	SECTION("Throws ConfigHandlerException if second param can't be parsed as filter expression") {
+		REQUIRE_THROWS_AS(ignores.handle_action(action, {"url", "!?"}),
+			ConfigHandlerException);
+	}
+}
+
+TEST_CASE("RssIgnores::handle_action() handles `always-download`",
+	"[RssIgnores]")
+{
+	RssIgnores ignores;
+
+	const std::string action = "always-download";
+
+	SECTION("Doesn't throw, regardless of how many params are passed") {
+		REQUIRE_NOTHROW(ignores.handle_action(action, {}));
```

Comment:
Not important but I would expect a `ConfigHandlerException` here (with `ActionHandlerStatus::TOO_FEW_PARAMS`).

Reply:
My goal here was to test what we already have, but I think your expectation is correct, so I'll make another PR. Don't want to add it to this one because that's a separate change and it may attract more reviewers.

Reply:
Done: #856.